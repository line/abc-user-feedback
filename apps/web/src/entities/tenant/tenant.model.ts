/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { create } from 'zustand';

import { client } from '@/shared';

import type { Tenant } from './tenant.type';

type State = { tenant: Tenant | null };

type Action = {
  setTenant: (tenant: Tenant) => void;
  refetchTenant: () => Promise<void>;
};

export const useTenantStore = create<State & Action>((set) => ({
  tenant: null,
  setTenant: (tenant) => set({ tenant }),
  refetchTenant: async () => {
    const { data } = await client.get({ path: '/api/admin/tenants' });
    set({ tenant: data });
  },
}));
