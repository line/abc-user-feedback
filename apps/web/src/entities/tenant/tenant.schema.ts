/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { z } from 'zod';

export const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z
    .array(z.string().refine((v) => /[a-z]+\.[a-z]{2,3}/.test(v)))
    .nullable(),
  useEmailVerification: z.boolean(),
  oauthConfig: z
    .object({
      clientId: z.string(),
      emailKey: z.string(),
      scopeString: z.string(),
      clientSecret: z.string(),
      authCodeRequestURL: z.string(),
      accessTokenRequestURL: z.string(),
      userProfileRequestURL: z.string(),
    })
    .nullable(),
});

export const tenantInfoSchema = tenantSchema.pick({
  id: true,
  siteName: true,
  description: true,
});
export const authInfoScema = tenantSchema.pick({
  isPrivate: true,
  isRestrictDomain: true,
  allowDomains: true,
  useOAuth: true,
  useEmail: true,
  oauthConfig: true,
});
