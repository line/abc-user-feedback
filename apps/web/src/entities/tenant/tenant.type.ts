/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { z } from 'zod';

const oauthConfigSchema = z.object({
  oauthUse: z.boolean(),
  clientId: z.string(),
  clientSecret: z.string(),
  authCodeRequestURL: z.string(),
  scopeString: z.string(),
  accessTokenRequestURL: z.string(),
  userProfileRequestURL: z.string(),
  emailKey: z.string(),
});

const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z.array(z.string()),
  useEmailVerification: z.boolean(),
  oauthConfig: oauthConfigSchema.nullable(),
});

export type Tenant = z.infer<typeof tenantSchema>;

export type OAuthConfig = z.infer<typeof oauthConfigSchema>;
