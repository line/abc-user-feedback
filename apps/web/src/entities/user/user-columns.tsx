/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { createColumnHelper } from '@tanstack/react-table';
import dayjs from 'dayjs';

import { Badge } from '@ufb/ui';

import { DATE_TIME_FORMAT, displayString, TableCheckbox } from '@/shared';

import UpdateUserPopover from './ui/update-user-popover.ui';
import type { UserMember } from './user.type';

const columnHelper = createColumnHelper<UserMember>();

export const getUserColumns = () => [
  columnHelper.display({
    id: 'select',
    header: ({ table }) => (
      <TableCheckbox
        checked={table.getIsAllRowsSelected()}
        indeterminate={table.getIsSomeRowsSelected()}
        onChange={table.getToggleAllRowsSelectedHandler()}
      />
    ),
    cell: ({ row }) => (
      <TableCheckbox
        checked={row.getIsSelected()}
        disabled={!row.getCanSelect()}
        indeterminate={row.getIsSomeSelected()}
        onChange={row.getToggleSelectedHandler()}
      />
    ),
    size: 50,
    enableSorting: false,
  }),
  columnHelper.accessor('email', {
    header: 'Email',
    enableSorting: false,
    size: 220,
  }),
  columnHelper.accessor('type', {
    header: 'Type',
    enableSorting: false,
    size: 80,
  }),
  columnHelper.accessor('name', {
    header: 'Name',
    enableSorting: false,
    cell: ({ getValue }) => displayString(getValue()),
    size: 120,
  }),
  columnHelper.accessor('department', {
    header: 'Department',
    cell: ({ getValue }) => displayString(getValue()),
    enableSorting: false,
    size: 120,
  }),
  columnHelper.accessor('members', {
    header: 'Project',
    cell: ({ getValue }) =>
      getValue().length > 0 ?
        <div className="flex flex-wrap gap-2">
          {getValue().map((member) => (
            <Badge key={member.id} type="secondary">
              {member.role.project.name}
            </Badge>
          ))}
        </div>
      : '-',
    enableSorting: false,
  }),
  columnHelper.accessor('createdAt', {
    header: 'Created',
    cell: ({ getValue }) => dayjs(getValue()).format(DATE_TIME_FORMAT),
    enableSorting: true,
  }),
  columnHelper.display({
    id: 'edit',
    header: 'Edit',
    cell: ({ row }) => <UpdateUserPopover user={row.original} />,
    size: 60,
  }),
];
