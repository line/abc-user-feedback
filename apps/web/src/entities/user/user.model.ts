/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import router from 'next/router';
import axios from 'axios';
import dayjs from 'dayjs';
import type { JwtPayload } from 'jwt-decode';
import { jwtDecode } from 'jwt-decode';
import { create } from 'zustand';

import { client, Path, sessionStorage } from '@/shared';
import type { Jwt } from '@/shared/types';

import type { User } from './user.type';

type State = { user: User | null };

type Action = {
  signInWithEmail: (input: {
    email: string;
    password: string;
  }) => Promise<void>;
  signInWithOAuth: (input: {
    code: string;
    callback_url?: string;
  }) => Promise<void>;
  signOut: () => Promise<void>;
  setUser: () => void;
  _signIn: (jwt: Jwt) => void;
};

export const useUserStore = create<State & Action>((set, get) => ({
  user: null,
  signInWithEmail: async ({ email, password }) => {
    const { data: jwt } = await axios.post('/api/login', { email, password });
    get()._signIn(jwt);
  },
  signInWithOAuth: async ({ code, callback_url }) => {
    const { data: jwt } = await axios.post('/api/oauth', {
      code,
      callback_url,
    });
    get()._signIn(jwt);
  },
  signOut: async () => {
    await axios.get('/api/logout');
    sessionStorage.removeItem('jwt');
    router.reload();
  },
  setUser: async () => {
    const jwt = sessionStorage.getItem('jwt');
    if (!jwt) return;

    const { sub, exp } = jwtDecode<JwtPayload>(jwt.accessToken);
    if (!sub || dayjs().isBefore(dayjs(exp))) get().signOut();
    else {
      const { data } = await client.get({
        path: '/api/admin/users/{id}',
        pathParams: { id: parseInt(sub) },
        options: { headers: { Authorization: `Bearer ${jwt.accessToken}` } },
      });
      set({ user: data });
    }
  },

  async _signIn(jwt) {
    sessionStorage.setItem('jwt', jwt);
    get().setUser();
    if (router.query.callback_url) {
      router.push(router.query.callback_url as string);
    } else {
      router.push({ pathname: Path.MAIN });
    }
  },
}));
