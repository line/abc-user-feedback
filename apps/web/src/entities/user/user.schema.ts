/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { z } from 'zod';

export const userSchema = z.object({
  id: z.number(),
  email: z.string().email(),
  type: z.union([z.literal('SUPER'), z.literal('GENERAL')]),
  name: z.string().nullable(),
  department: z.string().nullable(),
  signUpMethod: z.union([z.literal('OAUTH'), z.literal('EMAIL')]),
});

export const memberSchema = z.object({
  id: z.number(),
  role: z.object({
    name: z.string(),
    project: z.object({
      id: z.number(),
      name: z.string(),
    }),
  }),
});

export const userMemberSchema = userSchema.merge(
  z.object({
    members: z.array(memberSchema),
    createdAt: z.string(),
  }),
);

export const updateUserSchema = userSchema.omit({
  id: true,
  signUpMethod: true,
});
