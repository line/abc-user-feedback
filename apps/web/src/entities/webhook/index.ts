export * from './ui';
export * from './webhook.type';
