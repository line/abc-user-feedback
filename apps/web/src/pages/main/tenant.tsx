/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import { parseAsString, useQueryState } from 'nuqs';

import { Icon, Menu, MenuItem } from '@ufb/react';

import type { NextPageWithLayout } from '@/shared';
import SideMenuLayout from '@/shared/ui/side-menu-layout.ui';
import { Layout } from '@/widgets/layout';
import {
  LoginSetting,
  TenantInfoSetting,
  UserManagementSetting,
} from '@/widgets/setting-menu';

import serverSideTranslations from '@/server-side-translations';

const TenantPage: NextPageWithLayout = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const [currentMenu] = useQueryState<string>(
    'menu',
    parseAsString.withDefault('tenant'),
  );

  return (
    <SideMenuLayout
      sideMenu={
        <Menu
          type="single"
          orientation="vertical"
          className="w-full p-0"
          value={currentMenu}
          onValueChange={(value) =>
            router.push({ pathname: router.pathname, query: { menu: value } })
          }
        >
          <MenuItem value="tenant">
            <Icon name="RiInformation2Line" />
            {t('tenant-setting-menu.tenant-info')}
          </MenuItem>
          <MenuItem value="login">
            <Icon name="RiUser2Line" />
            {t('tenant-setting-menu.sign-up-mgmt')}
          </MenuItem>
          <MenuItem value="user">
            <Icon name="RiShieldKeyholeLine" />
            {t('tenant-setting-menu.user-mgmt')}
          </MenuItem>
        </Menu>
      }
    >
      {currentMenu === 'tenant' && <TenantInfoSetting />}
      {currentMenu === 'login' && <LoginSetting />}
      {currentMenu === 'user' && <UserManagementSetting />}
    </SideMenuLayout>
  );
};

TenantPage.getLayout = (page) => {
  return <Layout title="Tenant">{page}</Layout>;
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale)),
    },
  };
};

export default TenantPage;
