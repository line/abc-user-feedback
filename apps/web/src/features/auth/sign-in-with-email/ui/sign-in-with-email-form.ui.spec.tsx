/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { faker } from '@faker-js/faker';
import userEvent from '@testing-library/user-event';

import type { Tenant } from '@/entities/tenant';
import { useTenantStore } from '@/entities/tenant';
import * as user from '@/entities/user';

import { render, screen, waitFor } from '@/test-utils';
import SignInWithEmailForm from './sign-in-with-email-form.ui';

jest.mock('@/entities/user');

const DEFAULT_TENANT: Tenant = {
  allowDomains: [],
  isRestrictDomain: false,
  description: null,
  id: 1,
  oauthConfig: null,
  siteName: 'siteName',
  useEmail: true,
  useEmailVerification: true,
  useOAuth: true,
  isPrivate: false,
};

describe('SignInWithEmailForm', () => {
  beforeEach(() => {
    useTenantStore.setState({ tenant: DEFAULT_TENANT });
    jest.spyOn(user, 'useUserStore').mockReturnValue({
      signInWithEmail: jest.fn(),
      _signIn: jest.fn(),
      setUser: jest.fn(),
      signInWithOAuth: jest.fn(),
      signOut: jest.fn(),
    });
  });
  test('match snapshot when tenant is not private', () => {
    useTenantStore.setState({
      tenant: { ...DEFAULT_TENANT, isPrivate: false },
    });
    const { container } = render(<SignInWithEmailForm />);
    expect(container).toMatchSnapshot();

    expect(screen.getByText('button.sign-in')).toBeInTheDocument();

    expect(screen.queryByText('button.sign-up')).toBeInTheDocument();
  });
  test('match snapshot when tenant is private', () => {
    useTenantStore.setState({ tenant: { ...DEFAULT_TENANT, isPrivate: true } });
    const { container } = render(<SignInWithEmailForm />);
    expect(container).toMatchSnapshot();

    expect(screen.getByText('button.sign-in')).toBeInTheDocument();
    expect(screen.queryByText('button.sign-up')).not.toBeInTheDocument();
  });

  test('validation', async () => {
    render(<SignInWithEmailForm />);

    const signInBtn = screen.getByRole('button', {
      name: 'button.sign-in',
    });
    const idInput = screen.getByPlaceholderText('ID');
    const passwordInput = screen.getByPlaceholderText('Password');

    await userEvent.type(idInput, faker.internet.email());
    await userEvent.type(passwordInput, faker.string.alphanumeric(9));

    expect(signInBtn).not.toBeDisabled();

    await userEvent.clear(idInput);
    await userEvent.clear(passwordInput);

    await userEvent.type(idInput, faker.string.alphanumeric(8));
    await userEvent.type(passwordInput, faker.string.alphanumeric(8));

    expect(signInBtn).toBeDisabled();

    await userEvent.clear(idInput);
    await userEvent.clear(passwordInput);

    await userEvent.type(idInput, faker.internet.email());
    await userEvent.type(passwordInput, faker.string.alphanumeric(7));

    expect(signInBtn).toBeDisabled();
  });

  describe('Submittion', () => {
    test('on Success', async () => {
      jest.spyOn(user, 'useUserStore').mockImplementation(() => ({
        signInWithEmail: jest.fn(),
        _signIn: jest.fn(),
        setUser: jest.fn(),
        signInWithOAuth: jest.fn(),
        signOut: jest.fn(),
      }));
      render(<SignInWithEmailForm />);

      const idInput = screen.getByPlaceholderText('ID');
      const passwordInput = screen.getByPlaceholderText('Password');

      await userEvent.type(idInput, faker.internet.email());
      await userEvent.type(passwordInput, faker.string.alphanumeric(9));

      const submitBtn = screen.getByRole('button', {
        name: 'button.sign-in',
      });
      await userEvent.click(submitBtn);

      await waitFor(() =>
        expect(
          screen.getByText(new RegExp('success', 'i')),
        ).toBeInTheDocument(),
      );
    });
    test('on Error', async () => {
      jest.spyOn(user, 'useUserStore').mockImplementation(() => ({
        signInWithEmail: jest.fn().mockRejectedValue(new Error()),
        _signIn: jest.fn(),
        setUser: jest.fn(),
        signInWithOAuth: jest.fn(),
        signOut: jest.fn(),
      }));
      render(<SignInWithEmailForm />);

      const idInput = screen.getByPlaceholderText('ID');
      const passwordInput = screen.getByPlaceholderText('Password');

      await userEvent.type(idInput, faker.internet.email());
      await userEvent.type(passwordInput, faker.string.alphanumeric(9));
      const submitBtn = screen.getByRole('button', {
        name: 'button.sign-in',
      });
      await userEvent.click(submitBtn);

      await waitFor(() =>
        expect(screen.getByText(new RegExp('error', 'i'))).toBeInTheDocument(),
      );
    });
  });
});
