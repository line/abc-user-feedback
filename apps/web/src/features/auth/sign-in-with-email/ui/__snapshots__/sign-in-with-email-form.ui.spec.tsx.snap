// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInWithEmailForm match snapshot when tenant is not private 1`] = `
<div>
  <form>
    <div
      class="mb-6 flex flex-col gap-3"
    >
      <div
        class="flex flex-col gap-2"
      >
        <div
          class="relative"
        >
          <input
            class="input  input-lg pl-10 pr-10 "
            name="email"
            placeholder="ID"
            type="email"
          />
          <div
            class="absolute-y-center absolute left-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-tertiary"
              height="20"
              width="20"
            />
          </div>
          <div
            class="absolute-y-center absolute right-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-blue-primary"
              height="20"
              width="20"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <div
          class="relative"
        >
          <input
            class="input  input-lg pl-10 pr-10 "
            name="password"
            placeholder="Password"
            type="password"
          />
          <div
            class="absolute-y-center absolute left-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-tertiary"
              height="20"
              width="20"
            />
          </div>
          <div
            class="absolute-y-center absolute right-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-blue-primary"
              height="20"
              width="20"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-1"
    >
      <button
        class="btn btn-lg btn-primary"
        disabled=""
        type="submit"
      >
        button.sign-in
      </button>
      <button
        class="btn btn-lg btn-secondary"
        type="button"
      >
        button.sign-up
      </button>
    </div>
  </form>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;

exports[`SignInWithEmailForm match snapshot when tenant is private 1`] = `
<div>
  <form>
    <div
      class="mb-6 flex flex-col gap-3"
    >
      <div
        class="flex flex-col gap-2"
      >
        <div
          class="relative"
        >
          <input
            class="input  input-lg pl-10 pr-10 "
            name="email"
            placeholder="ID"
            type="email"
          />
          <div
            class="absolute-y-center absolute left-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-tertiary"
              height="20"
              width="20"
            />
          </div>
          <div
            class="absolute-y-center absolute right-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-blue-primary"
              height="20"
              width="20"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <div
          class="relative"
        >
          <input
            class="input  input-lg pl-10 pr-10 "
            name="password"
            placeholder="Password"
            type="password"
          />
          <div
            class="absolute-y-center absolute left-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-tertiary"
              height="20"
              width="20"
            />
          </div>
          <div
            class="absolute-y-center absolute right-[14px] flex items-center gap-2"
          >
            <svg
              class="inline-block text-blue-primary"
              height="20"
              width="20"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-1"
    >
      <button
        class="btn btn-lg btn-primary"
        disabled=""
        type="submit"
      >
        button.sign-in
      </button>
    </div>
  </form>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;
